xarray/core/dataarray.py

```diff
@@ -772,7 +780,18 @@ def encoding(self, value: Mapping[Hashable, Any]) -> None:
 
     @property
     def indexes(self) -> Indexes:
-        """Mapping of pandas.Index objects used for label based indexing"""
+        """Mapping of xarray Index or pandas.Index objects used for label based indexing."""
+        xr_or_pd_indexes = {}
+        for k, idx in self.xindexes.items():
+            try:
+                xr_or_pd_indexes[k] = idx.to_pandas_index()
+            except TypeError:
+                xr_or_pd_indexes[k] = idx
```

Comment:
I would drop the `try`/`except` clause here. Better to raise an error early rather than to have unexpected behavior.

Reply:
Agreed. Maybe we'll need a custom error message here to suggest users the `xindexes` property if what they're interested in getting all (new) xarray indexes?